diffmatchpatch/diff_test.go

```diff
@@ -998,8 +998,38 @@ func TestDiffText(t *testing.T) {
 }
 
 func TestDiffDelta(t *testing.T) {
+
+	type TestCase struct {
+		Text               string
+		Delta              string
+		ErrorMessagePrefix string
+	}
+
 	dmp := New()
 
+	// Error checking
+	for _, tc := range []TestCase{
+		{"jumps over the lazyx", "=4\t-1\t+ed\t=6\t-3\t+a\t=5\t+old dog", "Delta length (19) smaller than source text length (20)"},
+		{"umps over the lazy", "=4\t-1\t+ed\t=6\t-3\t+a\t=5\t+old dog", "Delta length (19) smaller than source text length (18)"},
+		{"", "+%c3%xy", "invalid URL escape \"%xy\""},
+		{"", "+%c3xy", "invalid UTF-8 token: \"\\xc3xy\""},
+		{"", "a", "Invalid diff operation in DiffFromDelta: a"},
+		{"", "-", "strconv.ParseInt: parsing \"\": invalid syntax"},
+		{"", "--1", "Negative number in DiffFromDelta: -1"},
+		{"", "", ""},
```

Comment:
While I agree with the rest of your comments, I'm struggling with this one. Initially I went without the empty case, but then I saw that it fits the pattern, and a similar case is handled similarly in [patch_test.go](https://github.com/sergi/go-diff/blob/master/diffmatchpatch/patch_test.go#L63) so I decided to move it in. I think the idea here is that checking for absence of error is also error checking, the other thing is the refactoring to avoid duplication.